src/app/socialMedia/social-media-list/social-media-list.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, OnInit } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+import { ApiService } from '../../services/api.service';
+import { SocialMedia } from '../../shared/social-media';
+import { SocialMediaType } from '../../shared/social-media-type';
+
+@Component({
+  selector: 'app-social-media-list',
+  templateUrl: './social-media-list.component.html',
+  styleUrls: ['./social-media-list.component.scss']
+})
+export class SocialMediaListComponent implements OnInit {
+
+  public socialmedia: SocialMedia[];
+  selectedSocialMedia: SocialMedia[];
+  public socialmediatypes: SocialMediaType[];
+  filterBy: string = 'All';
+
+  constructor(private router: Router, private apiService:  ApiService, private route: ActivatedRoute) { }
+
+  ngOnInit() {
+    
+    this.route.data.subscribe(data => {
+      this.socialmedia = data['socialmedia'];
+
+      // mockup data for the media types
+      for (var i=0; i<this.socialmedia.length;i++)
```

Comment:
Maybe we could use a foreach here instead, like socialMedia.forEach(); using a counted for loop can more easily lead to errors.